File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // prompts user for text
    string text = get_string("Enter a text: ");

    int letters = 0;
    int words = 1;
    int sentences = 0;

    // using a loop to count words, letters and sentences
    for (int i = 0; i < strlen(text); i++)
    {
        // counting the letters
        if (isalpha(text[i]))
        {
            letters++;
        }
        // counting the sentences
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
        // counting the words
        else if (isspace(text[i]))
        {
            words++;
        }
    }

    // calculate L and S
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;

    // round the index for the grades
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Input a text to be verified: ")

letters = 0
# the first word is not counted
words = 1
sentences = 0

i = 0
# len is strlen from C, but in Python
for i in range(len(text)):
    # check if the character is a letter
    if text[i].isalpha():
        letters += 1

    # checks for sentences
    elif text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1

    # checks for spaces to count words
    elif text[i].isspace():
        words += 1

# calculate L
# average number of letters per 100 words in the text
L = letters / words * 100

# calculate S
# average number of sentences per 100 words in the text
S = sentences / words * 100

# round the index for the grades
index = round(0.0588 * L - 0.296 * S - 15.8)

# print the grade level
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade ", + index)
